### PR TITLE
[integration test] Make message confirmed

### PR DESCRIPTION
**Pull request by acha-bill**
# Description of change

Tries to confirm the message on all peers provided by issues messages to build AW.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tools/integration-tests/tester/tests/testutil.go [hunk 1]
```diff
@@ -619,3 +619,52 @@ func txMetadataStateEqual(t *testing.T, node *framework.Node, txID string, expIn
...
+		if _, err := peer.GetMessage(msgID); err == nil {
+			peers = append(peers, peer)
+		}
+	}
+	if len(peers) == 0 {
+		log.Println("msg ", msgID, "is not found on any node")
+		t.FailNow()
+	}
+
+	confirmedOnAllPeers := func(msgID string, peers []*framework.Node) bool {
```
Comment (karimodm) [inline tools/integration-tests/tester/tests/testutil.go:636]: I think this method can be extracted as a general function to check if a certain message is confirmed on all supplied peers.

File: tools/integration-tests/tester/tests/testutil.go [hunk 2]
```diff
@@ -619,3 +619,52 @@ func txMetadataStateEqual(t *testing.T, node *framework.Node, txID string, expIn
 	}
 	return true, metadata.GradeOfFinality
 }
+
+// TryConfirmMessage tries to confirm the message on all the peers provided within the time limit provided.
+func TryConfirmMessage(t *testing.T, n *framework.Network, requiredPeers []*framework.Node, msgID string, waitFor time.Duration, tick time.Duration) {
+	var peers []*framework.Node
+	for _, peer := range n.Peers() {
```
Comment (karimodm) [inline tools/integration-tests/tester/tests/testutil.go:626]: Is there a use-case to collect the peers on which the message is available separately if we want the requiredPeers to have the message confirmed?

File: tools/integration-tests/tester/tests/testutil.go [hunk 3]
```diff
@@ -619,3 +619,52 @@ func txMetadataStateEqual(t *testing.T, node *framework.Node, txID string, expIn
...
+			if metadata.GradeOfFinality != gof.High {
+				return false
+			}
+		}
+		return true
+	}
+
+	var i int
+	timer := time.NewTimer(waitFor)
+	ticker := time.NewTicker(tick)
```
Comment (karimodm) [inline tools/integration-tests/tester/tests/testutil.go:649]: Release resources for this ticker with `defer ticker.Stop()`

**Review by karimodm — CHANGES_REQUESTED** [review]

**Comment by karimodm** [comment]
Thanks for the PR @acha-bill. Are you also gonna try and use the new API to more precisely confirm a certain message by specifying parents manually?

File: tools/integration-tests/tester/tests/testutil.go (hunk 2 above)
Comment (acha-bill) [reply tools/integration-tests/tester/tests/testutil.go:626]: yeah. It's a short circuit check to make sure that the message exists in the network in the first place.
Else, you'll wait for the whole timeout for nothing.

File: tools/integration-tests/tester/tests/testutil.go (hunk 3 above)
Comment (acha-bill) [reply tools/integration-tests/tester/tests/testutil.go:649]: ouch :)

**Comment by acha-bill** [comment]
> Thanks for the PR @acha-bill. Are you also gonna try and use the new API to more precisely confirm a certain message by specifying parents manually?

That API will be useful for branches/txs.
We could also use it here but there's no need IMO.

File: tools/integration-tests/tester/tests/common/common_synchronization_test.go [hunk 4]
```diff
@@ -133,6 +133,46 @@ func TestFirewall(t *testing.T) {
...
+	})
+	require.NoError(t, err)
+	defer tests.ShutdownNetwork(ctx, t, n)
+
+	peers := n.Peers()
+	msgID, err := peers[0].Data([]byte("test"))
+	require.Nil(t, err)
+	metadata, err := peers[0].GetMessageMetadata(msgID)
+	require.Nil(t, err)
+	log.Printf("gof of msg %s = %s", msgID, metadata.GradeOfFinality.String())
```
Comment (jkrvivian) [inline tools/integration-tests/tester/tests/common/common_synchronization_test.go:172]: Is it necessary to check gof before `TryConfirmMessage`?

Comment (acha-bill) [reply tools/integration-tests/tester/tests/common/common_synchronization_test.go:172]: Nope. Its just a debug log.

**Review by karimodm — APPROVED** [review]

**Pull request merged**